File: LeetCode/Easy/0027-remove-element/0027-remove-element.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int st = -1;
        // first occurrence
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == val) {
                st = i;
                break;
            }
        }

        // if val not found return full arr size
        if (st == -1) {
            return nums.size();
        }

        // remove all val with forward elems
        for (int j = st + 1; j < nums.size(); j++) {
            if (nums[j] != val) {
                nums[st] = nums[j];
                st++;
            }
        }
        return st;
    }
};

File: LeetCode/Easy/0027-remove-element/0027-remove-element.go
func removeElement(nums []int, val int) int {
    i:= 0
    for k := range nums {
        if nums[k] == val{
            break
        }
        i++
    }
    if i>=len(nums){
        return len(nums)
    }

    j:= i+1
    for j<len(nums){
        if nums[j] != val {
            // swap
            nums[i], nums[j] = nums[j], nums[i]
            i++
        }
        j++
    }
    return i
}